clean/platforms/nextrequest.py

```diff
@@ -0,0 +1,333 @@
+                docpath = profile["base_url"] + docpath
+            docsplit = parsed_docpath.path.split("/")
+            if (
+                len(docsplit) == 3
+                and docsplit[1] == "documents"
+                # and docsplit[2] == entry["id"]
+            ):
+                docpath += "/download?token="
+            if urlparse(docpath).scheme == "":
+                docpath = "https:" + docpath
+            line["asset_url"] = docpath
+
+            # au = entry["asset_url"]
+            # if urlparse(au).netloc == "":
+            #     au = profile["base_url"] + au
+            # line["asset_url"] = au
+            for item in ["subfolder_name", "folder_name"]:
+                if item in entry and len(entry[item]) > 0:
+                    folder_id = folder_id + "__" + entry[item]
+            line["case_id"] = folder_id
+        else:
+            logger.error(f"Do not understand sitetype {line['sitetype']}.")
+
+        line["case_id"] = folder_id
+        line["name"] = entry["title"]
+
+        # Use filename and local_cache's root directory to identify a path relative to the scraper's folder
+        partial_path = PurePath(filename).relative_to(local_cache.path)
+        partial_path = str(partial_path.relative_to(partial_path.parts[0]).as_posix())
+        line["parent_page"] = partial_path
+        line["title"] = entry["title"]
+
+        if "details" not in line:
+            line["details"] = {}
+
+        for target in profile["details"]:
+            source = profile["details"][target]
+            if "ds!" in source:
+                source = source.replace("ds!", "")
+                if source not in entry["document_scan"]:
+                    logger.debug(
+                        f"Missing ['document_scan']['{source}'] from entry {entry}"
+                    )
+                else:
+                    line["details"][target] = entry["document_scan"][source]
+            else:  # Straight shot, no subkey
+                if source not in entry:
+                    logger.warning(f"Missing {source} from entry {entry}")
+                else:
+                    line["details"][target] = entry[source]
+        local_metadata.append(line)
+    return local_metadata
+
+
+def fingerprint_nextrequest(start_url: str):
+    """
+    Given a link to a NextRequest documents folder, try to ID how the site stores stuff.
+
+    Args:
+        start_url (str): The web page for the folder of NextRequest docs you want
+    Returns:
+        local_schema (dict)
+    """
+    """To-do:
+        Parser needs to map out all of the metadata locations we want to preserve.
+        Where is the asset URL?
+        What details do we want to preserve?
+        How are the references to do those details stored, like how do we handle the subkey?
+        Who is we, and why is we typing this at 10 p.m.?
+    """
+    line = None
+    parsed_url = urlparse(start_url)
+    if parsed_url.path == "/documents":  # LAPDish type
```

Comment:
Suggestion: let's isolate the NextRequest checks into a separate function rather than spread across if/elifs. That'll allow you to document different types a bit better in one place, but I am cognizant of multiplying variations. If that function becomes unwieldy, I'd move it into a separate file for readability.

Reply:
I'm confused. I was thinking that you were maybe talking about like a parse_nextrequest wrapper function that might do the fingerprinting, then call, say, parse_nextrequest_lapdish or parse_nextrequest_bartish based on the fingerprinting. 

But you're actually flagging the code in the fingerprinting process itself. Should I split that into separate workflows to get separate returns based on the approach described in my previous paragraph? Should I also split up the parsing that way?

Reply:
Apologies, that was unclear. After re-reading the code, I understand it better. The suggestion was to assist with additional fingerprints by creating a smaller, centralized place—either a dictionary of `fingerprints` or a function like `get_fingerprint`—which contains the specific logic bits/flags for each NextRequest type that gets used by the `fingerprint_nextrequest`.

```py
def get_fingerprint(parsed_url):
    fingerprint = None
    if parsed_url.path == "/documents":
        fingerprint = "lapdish" # can there be multiple fingerprints?
    elif (
        len(parsed_url.path.split("/")) == 3
        and parsed_url.path.split("/")[1] == "requests"
    ):
        fingerprint = "bartish"
    return fingerprint


def fingerprint_nextrequest():
    ...
    fingerprint = get_fingerprint(parsed_url)
    if fingerprint == "lapdish":
        line = {
    elif fingerprint == "bartish"
        ...
```